libp2p/Host.cpp

```diff
@@ -24,6 +24,7 @@
 #include <chrono>
 #include <thread>
 #include <mutex>
+#include <boost/asio.hpp>
```

Comment:
windows.h is included (indirectly) in only a few places in the codebase - could we have a more targeted set of asio includes, together with a line explaining why it is there in each case?